subsys/zigbee/Kconfig

```diff
@@ -18,6 +19,7 @@ menuconfig ZIGBEE
 	imply FLASH_PAGE_LAYOUT
 	imply FLASH_MAP
 	imply MPU_ALLOW_FLASH_WRITE
+	imply NETWORKING
```

Comment:
Why do we imply networking? If zigbee requires networking to work it should be "select" in case you want it to enable when enabling zigbee or "depends on" if you want to be able to enable zigbee only when networking enabled (note depends on is preferred).  If it is still possible to work with zigbee with networking disable this seems to be ok. @rlubos what do you think?

Reply:
Zigbee won't work without `NETWORKING` so it's a necessary component. The reason I used `IMPLY` is because `NETWORKING` is a visible symbol in menuconfig i.e. it has prompt, some docs say one should not use `SELECT` for visible symbols.
However @MarekPorwisz's argument also seems valid. @rlubos what is your opinion?

Reply:
I think `select` is fine for this particular case, given that `NETWORKING` is a hard-dependency for zigbee now. If you prefer `depends on`, it fine as well, but keep in mind this will require to add `CONFIG_NETWORKING=y` to every single sample.

Reply:
Changed to `SELECT`. Also, `NRF_802154_RADIO_DRIVER` removed as it's selected by `IEEE802154_NRF5`